packages/gatsby-theme-newrelic/src/components/ResultView.js

```diff
@@ -0,0 +1,128 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {
+  appendClassName,
+  getUrlSanitizer,
+} from '@elastic/react-search-ui-views/es/view-helpers';
+import { isFieldValueWrapper } from '@elastic/react-search-ui-views/es/types/FieldValueWrapper';
+import { css } from '@emotion/core';
+import { rgba } from 'polished';
+
+function getFieldType(result, field, type) {
+  if (result[field]) return result[field][type];
+}
+
+function getRaw(result, field) {
+  return getFieldType(result, field, 'raw');
+}
+
+function getSnippet(result, field) {
+  return getFieldType(result, field, 'snippet');
+}
+
+function htmlEscape(str) {
+  if (!str) return '';
+
+  return String(str)
+    .replace(/&/g, '&amp;')
+    .replace(/"/g, '&quot;')
+    .replace(/'/g, '&#39;')
+    .replace(/</g, '&lt;')
+    .replace(/>/g, '&gt;');
+}
+
+function getEscapedField(result, field) {
+  // Fallback to raw values here, because non-string fields
+  // will not have a snippet fallback. Raw values MUST be html escaped.
+  const safeField =
+    getSnippet(result, field) || htmlEscape(getRaw(result, field));
+  return Array.isArray(safeField) ? safeField.join(', ') : safeField;
+}
+
+function getEscapedFields(result) {
+  return Object.keys(result).reduce((acc, field) => {
+    // If we receive an arbitrary value from the response, we may not properly
+    // handle it, so we should filter out arbitrary values here.
+    //
+    // I.e.,
+    // Arbitrary value: "_metaField: '1939191'"
+    // vs.
+    // FieldValueWrapper: "_metaField: {raw: '1939191'}"
+    if (!isFieldValueWrapper(result[field])) return acc;
+    return { ...acc, [field]: getEscapedField(result, field) };
+  }, {});
+}
+
+function ResultView({
+  className,
+  result,
+  onClickLink,
+  titleField,
+  urlField,
+  ...rest
+}) {
+  const fields = getEscapedFields(result);
+  const title = getEscapedField(result, titleField);
+  const url = getUrlSanitizer(URL, window.location)(getRaw(result, urlField));
+  const newRelicSite = fields.url.split('.newrelic')[0].slice(8);
```

Comment:
What does this `slice(8)` do? We might want to add some comments to make this a little more clear.

Reply:
this is slicing off the `https://` of the url so we can just get the sub domain, can definitely add a comment in!